### PR TITLE
Adapted Observation Request Creation

### PR DESCRIPTION
**Pull request by btw04**
The observation creation view now dynamically checks whether the passed parameters are flat structured. 
~~This PR is currently missing tests using url-encoded parameters and malformed flat structures. It mainly aims to discuss the naming and mapping, so please check if this aligns with your expectations @SprangerjS~~

It also includes some changes to the models and further validates input data.

### CODE REVIEW

**Comment by btw04** [comment]
Seems like the Django requests are automatically url-form encoded: https://stackoverflow.com/questions/43166983/django-x-www-form-urlencoded-request

**Comment by btw04** [comment]
This PR combines https://trello.com/c/LN1uVFS2, https://trello.com/c/BBlmZ2Vy and https://trello.com/c/KUIsz2GQ

**Comment by btw04** [comment]
This PR is open for review, but I am waiting for final feedback from Robert regarding the bound values

File: observation_data/data_verification.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    """
+    Verify data integrity.
+    :param name: Name of the attribute
+    :param value: Value of the attribute
+    :return: Error if the value is invalid or None
+    """
+    if isinstance(value, dict) or isinstance(value, list):
+        return
+
+    if name == "frames_per_filter":
```
Comment (bennetbo) [inline observation_data/data_verification.py:69]: Could we use a switch statement here instead? That way you don't have to write `name ==` 20 times

File: observation_data/tests.py [hunk 2]
```diff
@@ -19,35 +21,50 @@
 )
 
 
+def _create_user_and_login(test_instance):
+    call_command("generate_admin_user")
+    load_dotenv()
+    admin_username = os.getenv("ADMIN_EMAIL")
```
Comment (bennetbo) [inline observation_data/tests.py:27]: `generate_admin_user` wont work when this is not set, so we could just move this check to the start of the function

File: TURMFrontend/settings.py [hunk 3]
```diff
@@ -194,3 +195,6 @@ def get_env_var(var_name):
         },
     },
 }
+
+if "test" in sys.argv:
```
Comment (bennetbo) [inline TURMFrontend/settings.py:199]: Have not tested this myself, but would't it be useful to see error logs in tests?

**Review by bennetbo — APPROVED** [review]

**Comment by bennetbo** [comment]
LGTM, not sure about disabling logging in tests

**Comment by btw04** [comment]
> LGTM, but not sure if we should disable logging in tests

The issue is that my tests deliberately send bad requests, which are logged as console output during the test runs. Maybe other tests can enable it?

**Pull request merged**